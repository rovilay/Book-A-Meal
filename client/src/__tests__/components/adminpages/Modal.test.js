import React from 'react';
import { shallow } from 'enzyme';
import ModalComp from '../../../components/adminpages/Modal/Index';
import { menuMeals, menus } from '../../__mockData__/menuMock';


const props = {
  modal: {
    isOpen: true,
    isEdit: true,
    isOrderInfo: true,
    contentLabel: 'Modal',
    content: {
      menuId: menus[0].id,
      meals: menuMeals.slice(0, 2)
    }
  },
  meals: menuMeals,
  deleteMealInEditMenu: jest.fn(),
  editMenuMeals: jest.fn(),
  addMealInEditMenu: jest.fn(),
  updateMenu: jest.fn(),
  orderedMealsPagination: {},
  setModal: jest.fn(),
  hideModal: jest.fn(),
  emptyEditMenu: jest.fn(),
};

describe('ModalComp component test', () => {
  const setup = () => shallow(<ModalComp {...props} />);


  it('should render ModalComp correctly', (done) => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();

    done();
  });
});
