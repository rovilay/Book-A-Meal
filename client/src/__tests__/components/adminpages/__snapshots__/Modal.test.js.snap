// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalComp component test should render \`Modal Component\` correctly 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={0}
  contentLabel="Modal"
  isOpen={true}
  overlayClassName="overlay"
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div
    className="closeBtn"
  >
    <button
      className=""
      onClick={[MockFunction]}
    >
      <FontAwesome
        name="times"
        size="2x"
      />
    </button>
  </div>
  <EditMenuTable
    addMealInEditMenu={[MockFunction]}
    content={
      Object {
        "meals": Array [
          Object {
            "id": "1",
            "title": "EBA AND EGUSI",
          },
          Object {
            "id": "2",
            "title": "SALAD",
          },
        ],
        "menuId": "1",
      }
    }
    deleteMealInEditMenu={[MockFunction]}
    editMenuMeals={[MockFunction]}
    emptyEditMenu={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    hideModal={[MockFunction]}
    isEdit={true}
    meals={
      Array [
        Object {
          "id": "1",
          "title": "EBA AND EGUSI",
        },
        Object {
          "id": "2",
          "title": "SALAD",
        },
        Object {
          "id": "3",
          "title": "SPAGHETTI",
        },
        Object {
          "id": "4",
          "title": "YAM AND EGG",
        },
      ]
    }
    modal={
      Object {
        "content": Object {
          "meals": Array [
            Object {
              "id": "1",
              "title": "EBA AND EGUSI",
            },
            Object {
              "id": "2",
              "title": "SALAD",
            },
          ],
          "menuId": "1",
        },
        "contentLabel": "Modal",
        "isEdit": true,
        "isOpen": true,
        "isOrderInfo": true,
      }
    }
    orderedMealsPagination={Object {}}
    setModal={[MockFunction]}
    title="Modal"
    updateMenu={[MockFunction]}
  />
  <OrderDetailsTable
    addMealInEditMenu={[MockFunction]}
    content={
      Object {
        "meals": Array [
          Object {
            "id": "1",
            "title": "EBA AND EGUSI",
          },
          Object {
            "id": "2",
            "title": "SALAD",
          },
        ],
        "menuId": "1",
      }
    }
    deleteMealInEditMenu={[MockFunction]}
    editMenuMeals={[MockFunction]}
    emptyEditMenu={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    hideModal={[MockFunction]}
    isEdit={true}
    meals={
      Array [
        Object {
          "id": "1",
          "title": "EBA AND EGUSI",
        },
        Object {
          "id": "2",
          "title": "SALAD",
        },
        Object {
          "id": "3",
          "title": "SPAGHETTI",
        },
        Object {
          "id": "4",
          "title": "YAM AND EGG",
        },
      ]
    }
    modal={
      Object {
        "content": Object {
          "meals": Array [
            Object {
              "id": "1",
              "title": "EBA AND EGUSI",
            },
            Object {
              "id": "2",
              "title": "SALAD",
            },
          ],
          "menuId": "1",
        },
        "contentLabel": "Modal",
        "isEdit": true,
        "isOpen": true,
        "isOrderInfo": true,
      }
    }
    orderedMealsPagination={Object {}}
    setModal={[MockFunction]}
    title="Modal"
    updateMenu={[MockFunction]}
  />
</Modal>
`;
